### PR TITLE
pythonPackages.searx: 0.8.1 -> 0.9.0

### PR DESCRIPTION
**Pull request by fpletz**
###### Things done
- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -16781,12 +16795,13 @@ in modules // {
...
+  pyasn1 = buildPythonPackage rec {
+    name = "pyasn1-${version}";
+    version = "0.1.9";
 
     src = pkgs.fetchurl {
-      url = "mirror://sourceforge/pyasn1/0.1.8/pyasn1-0.1.8.tar.gz";
-      sha256 = "0iw31d9l0zwx35szkzq72hiw002wnqrlrsi9dpbrfngcl1ybwcsx";
+      url = "mirror://sourceforge/pyasn1/${version}/pyasn1-${version}.tar.gz";
+      sha256 = "0zraxni14bqi20kr4bi6nwsh32aibz0fq0xaczfisw0zdpcsqg45";
     };
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: could you use the pypi mirror here
https://pypi.python.org/pypi/pyasn1/0.1.9

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -25136,16 +25151,17 @@ in modules // {
...
-    name = "searx-0.8.1";
+    name = "searx-${version}";
+    version = "0.9.0";
 
     src = pkgs.fetchurl {
-      url = "https://github.com/asciimoo/searx/archive/v0.8.1.tar.gz";
-      sha256 = "0z0s9n8iblrw7y5xrh2apzsazkgm4vzmxn0ckw4yfiya9am8zk32";
+      url = "https://github.com/asciimoo/searx/archive/v${version}.tar.gz";
+      sha256 = "0hbwvyah7mml452fyr7j14dqsbp1xsk7dglqpvsgg5cq0aiq9yp5";
     };
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: same here

**Comment by FRidh** [comment]
Looks good to me.

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (fpletz) [inline pkgs/top-level/python-packages.nix]: Unfortunately, the current version of searx is not yet on pypi. But I'll convert that to `fetchFromGitHub`.

Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: https://pypi.python.org/pypi/searx/0.9.0 ?

Comment (fpletz) [inline pkgs/top-level/python-packages.nix]: Oh, I was searching here: https://pypi.python.org/pypi?%3Aaction=search&term=searx&submit=search

Thanks, will do.

Comment (fpletz) [inline pkgs/top-level/python-packages.nix]:
```
building path(s) ‘/nix/store/0kdv4m6k8jbdp2wnvd8pp4xmirri9c8m-python2.7-searx-0.9.0’
unpacking sources
unpacking source archive /nix/store/ghdzg3c785zx2ik6jd5ad4f43m7fw4kp-searx-0.9.0.tar.gz
source root is searx-0.9.0
setting SOURCE_DATE_EPOCH to timestamp 1464090064 of file searx-0.9.0/setup.cfg
patching sources
configuring
building
Traceback (most recent call last):
  File "nix_run_setup.py", line 6, in <module>
    exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
  File "setup.py", line 20, in <module>
    requirements = map(str.strip, open('requirements.txt').readlines())
IOError: [Errno 2] No such file or directory: 'requirements.txt'
builder for ‘/nix/store/gnhcvdvgqf4a8brnizh3rfl7amzgpygi-python2.7-searx-0.9.0.drv’ failed with exit code 1
```

Now I'm confused. The setup.py is opening requirements.txt but that's not in the tarball at pypi. :frowning:

Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: Opened issue upstream https://github.com/asciimoo/searx/issues/572

**Pull request merged**